relay-server/src/actors/envelopes.rs

```diff
@@ -1872,15 +1873,20 @@ impl Handler<ProcessMetrics> for EnvelopeProcessor {
         for item in items {
             let payload = item.payload();
             if item.ty() == ItemType::Metrics {
-                let timestamp = item.timestamp().unwrap_or(default_timestamp);
-                let metrics = Metric::parse_all(&payload, timestamp).filter_map(|result| {
-                    let mut metric = result.ok()?;
-                    clock_drift_processor.process_timestamp(&mut metric.timestamp);
-                    Some(metric)
-                });
-
-                relay_log::trace!("inserting metrics into project cache");
-                project_cache.do_send(InsertMetrics::new(public_key, metrics));
+                let mut timestamp = item.timestamp().unwrap_or(received_timestamp);
+                clock_drift_processor.process_timestamp(&mut timestamp);
+
+                let min_timestamp =
+                    max(0, received.timestamp() - self.config.max_secs_in_past()) as u64;
+                let max_timestamp =
+                    (received.timestamp() + self.config.max_secs_in_future()) as u64;
+                if min_timestamp <= timestamp.as_secs() && timestamp.as_secs() <= max_timestamp {
+                    let metrics =
+                        Metric::parse_all(&payload, timestamp).filter_map(|result| result.ok());
```

Comment:
This section concerns raw metrics, not aggregated metrics buckets, but I figured it makes sense to check the timestamps here as well. @jan-auer is it OK to use the same range as for error events (30 days in the past, 1 minute in the future)?